utilities/src/main/java/io/syndesis/qe/bdd/CommonSteps.java

```diff
@@ -137,4 +137,27 @@ public void resetState() {
     public void sleepForJenkinsDelay(int secs) {
         TestUtils.sleepForJenkinsDelayIfHigher(secs);
     }
+
+    @Then("^wait for Todo to become ready")
+    public void waitForTodo() {
+        log.info("Waiting for Todo to get ready");
+        ExecutorService executorService = Executors.newFixedThreadPool(1);
+        Runnable runnable = () ->
+                    OpenShiftUtils.xtf().waiters()
+                            .areExactlyNPodsReady(1, "syndesis.io/app", "todo")
+                            .interval(TimeUnit.SECONDS, 20)
+                            .timeout(TimeUnit.MINUTES, 12)
+                            .assertEventually();
+            executorService.submit(runnable);
+
+        executorService.shutdown();
+        try {
+            if (!executorService.awaitTermination(20, TimeUnit.MINUTES)) {
+                executorService.shutdownNow();
+                fail("Todo app wasn't initilized in time");
+            }
+        } catch (InterruptedException e) {
+            fail("Todo app wasn't initilized in time");
```

Comment:
the reason for this failure is that the wait was interrupted, right? I'd change the message, can you also add the actual exception to the fail method?

Reply:
I don't know. I just used the code from 'waiting for syndesis' part. I will update it.